### PR TITLE
Added basic READ THE DOCS.

### PR DESCRIPTION
**Pull request by Jeff-Wang93**
Added a new folder called docs in the project directory. Within there, all the relevant `.rst` files are located in source. 

I also edited the project `Makefile` to auto generate the RST files within `doc/source/`.

I also added some comments to /api/plugins.py. Not close to being finished with the documentation though. 

Current problems:  Sphinx doesn't like files with `-` in them. I'll add a little thing to the Makefile that changes any file/folder with a `-` in it to a `_`, build the docs, and then revert the names back to what they originally were.

### CODE REVIEW

**Comment by Jeff-Wang93** [comment]
need to fix Travis to include sphinx

**Comment by cglewis** [comment]
I'm ok with the linters/checkers ignoring the `docs` folder

**Comment by cglewis** [comment]
as for sphinx, I assume you're talking about the python library? just add it to the `requirements.txt` in the tests folder

**Comment by Jeff-Wang93** [comment]
Need to fix the code climate formatting stuff.

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+#
+# source_suffix = ['.rst', '.md']
+source_suffix = '.rst'
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = u'Vent'
+copyright = u'2017, Cyber Reboot'
```
Comment (cglewis) [inline docs/source/conf.py]: This should be `2017, In-Q-Tel, Inc.`

**Comment by Jeff-Wang93** [comment]
Once I get the Makefile script working, I'll take away the [WIP] tag.

**Comment by Jeff-Wang93** [comment]
When make is called, it will change the names of the directories in `vent/core`. 
It will then call autodoc to create the RST files, and it'll change the names back. 
I also change the name of `__init__.py` since it gets changed too. I could probably just not use `_` to change names. 

I also need to make it so that all `-` are changed. `rmq-es-connector` has 2 and my little code only changes the first `-` so Sphinx can't correctly document it.

**Comment by Jeff-Wang93** [comment]
I actually dont think my little code will work with Travis. Will look into it more.

**Comment by Jeff-Wang93** [comment]
I'd be grateful if you could check how correct my little bash script in the makefile is.

**Comment by cglewis** [comment]
@Jeff-Wang93 seems to do the job, provided a couple tips on making it cleaner.

File: .codeclimate.yml [hunk 2]
```diff
@@ -16,3 +16,4 @@ ratings:
   - "**.py"
 exclude_paths:
 - tests/
+- docs/
```
Comment (cglewis) [inline .codeclimate.yml]: alphabetize, please.

**Pull request merged**